keps/sig-api-machinery/2144-clientgo-apply/README.md

```diff
@@ -358,6 +284,66 @@ adding functions to interfaces is a change we have made in the past, and develop
 that have alternate implementations of the interface will usually get a compiler
 error in this case, which is relatively trivial to resolve
 
+### read/modify/write loop support
+
+While it is
+[recommended](https://kubernetes.io/docs/reference/using-api/server-side-apply/)
+that apply clients use "fully specified intent" rather than
+read/modify/write loops, there are many existing controllers that use
+a get/modify/update loop today, and are not easy to convert to the
+"fully specified intend" approach.
+
+For such controllers, providing a convenient way to do a
+"get-previously-applied/modify/apply" loop is pragmatic. It allows
+these controller to benefit from apply by sending a minimal apply
+patch (as opposed to sending the entire object for update) and to
+reduce the odds of conflict with other controllers.
+
+To support this, the builders will include "BuildApply" utility functions
+function. For example:
+
+```
+fieldManger := "my-field-manager"
+// 1. read
+deployment, err := client.AppsV1().Deployments(ns).Get("deployment-name", metav1.GetOptions{})
+if err != nil {
+  // handle err
+}
+applyConfig, err := appsv1apply.BuildDeploymentApply(deployment, fieldManager)
+if err != nil {
+  // handle err
+}
+
+// 2. modify
+applyConfig.GetSpec().Replicas(10)
+
+// 3. apply
+client.AppsV1().Deployments(ns).Apply(ctx, applyConfig, metav1.ApplyOptions{FieldManager: fieldManager, Force: true})
+```
+
+In the above example, `BuildDeploymentApply` constructs a populated
+apply configuration from a deployment object returned from a Get. It
+uses the provided field manager to get the matching field set
+(`FieldsV1` data) from object and then combines the field set with the
+object to produce the apply configuration.
+
+We will clearly document that on the "BuildApply" functions that we
+recommend using "fully specified intent" when using apply and also
+when it might be appropriate (and inappropriate) to use the "BuildApply"
+utility.
+
+An alternative we considered, but rejected, was to add `GetForApply()`
+style functions to the client. The benefit of this approach is that
+the caller doesn't have to deal with converting the response object
+into an apply configuration, and there is no possibility of the client
+modifying the object before converting it into an apply
+configuration. The problem with this approach is that it is tightly
+coupled, and there are many operations that return an object
```

Comment:
I don't personally care about the tight coupling but the second argument is more convincing.